### PR TITLE
Pie Chart

### LINKED ISSUE #22: [8][FE][RVS] Pie Chart of Commits Number of Each Contributor (GitHub/GitLab)
Draw pie chart of contributions

### PR DESCRIPTION
**Pull request by jn2657**
# Description

To show the contribution of each member in project, a pie chart with commit data is added.

# ScreenShot
截圖 2021-12-25 下午8 05 26

### CODE REVIEW

**Comment by Xanonymous-GitHub** [comment]
Please rebase this branch to the latest head in main branch

**Review by VincentETN — COMMENTED** [review]
處理 loading 的部分能不能分開，只出現在另一支 PR，我覺得不改的話，同樣的地方應該會全部 conflict。

File: src/App/component/CodeBasePage.jsx [hunk 1]
```diff
@@ -76,39 +97,65 @@ function CodeBasePage(prop) {
...
+
+  const getGitLabCommitFromDB = () => {
+    const gitlabRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'gitlab')
+    if(gitlabRepo !== undefined){
+      const query = gitlabRepo.url.split("gitlab.com/")[1]
+      Axios.get(`http://localhost:9100/pvs-api/gitlab/commits/${query}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then((response) => {
+        setCommitListData(previousArray =>[...previousArray, ...response.data])
+      })
```
Comment (Xanonymous-GitHub) [inline src/App/component/CodeBasePage.jsx]: Suggest check if the response.data is not null or undefined before using `...`
```suggestion
      .then((response) => {
        if (response?.data) {
          setCommitListData(previousArray =>[...previousArray, ...response.data])
        }
      })
```

File: src/App/component/CommitsPage.jsx [hunk 2]
```diff
@@ -55,65 +68,99 @@ function CommitsPage(prop) {
...
+    if (githubRepo !== undefined){
+      const query = githubRepo.url.split("github.com/")[1]
+      // todo need refactor with async
+      Axios.get(`http://localhost:9100/pvs-api/github/commits/${query}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then((response) => {
+        setCommitListData(response.data)
+      })
+      .catch((e) => {
+        alert(e.response.status)
```
Comment (Xanonymous-GitHub) [inline src/App/component/CommitsPage.jsx]:
```suggestion
        alert(e.response?.status)
```

File: src/App/component/CommitsPage.jsx [hunk 3]
```diff
@@ -55,65 +68,99 @@ function CommitsPage(prop) {
...
+    if(gitlabRepo !== undefined){
+      const query = gitlabRepo.url.split("gitlab.com/")[1]
+      Axios.post(`http://localhost:9100/pvs-api/gitlab/commits/${query}`, "",
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then(() => {
+        getGitLabCommitFromDB()
+        setLoading(false)
+      })
+      .catch((e) => {
+        alert(e.response.status)
```
Comment (Xanonymous-GitHub) [inline src/App/component/CommitsPage.jsx]:
```suggestion
        alert(e.response?.status)
```

File: src/App/component/ContributionPage.jsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+  // Get current project
+  useEffect(() => {
+    Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+      { headers: { "Authorization": `${jwtToken}` } })
+      .then((response) => {
+        setCurrentProject(response.data)
+      })
+      .catch((e) => {
+        alert(e.response.status)
```
Comment (Xanonymous-GitHub) [inline src/App/component/ContributionPage.jsx]:
```suggestion
        alert(e.response?.status)
```

File: src/App/component/ContributionPage.jsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      // todo need refactor with async
+      Axios.get(`http://localhost:9100/pvs-api/github/commits/${query}`,
+        { headers: { "Authorization": `${jwtToken}` } })
+        .then((response) => {
+          setCommitListData(response.data)
+        })
+        .catch((e) => {
+          alert(e.response.status)
```
Comment (Xanonymous-GitHub) [inline src/App/component/ContributionPage.jsx]:
```suggestion
          alert(e.response?.status)
```

File: src/App/component/ContributionPage.jsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    if (gitlabRepo !== undefined) {
+      const query = gitlabRepo.url.split("gitlab.com/")[1]
+      Axios.post(`http://localhost:9100/pvs-api/gitlab/commits/${query}`, "",
+        { headers: { "Authorization": `${jwtToken}` } })
+        .then(() => {
+          getGitLabCommitFromDB()
+          setLoading(false)
+        })
+        .catch((e) => {
+          alert(e.response.status)
```
Comment (Xanonymous-GitHub) [inline src/App/component/ContributionPage.jsx]:
```suggestion
          alert(e.response?.status)
```

File: src/App/component/CodeBasePage.jsx [hunk 7]
```diff
@@ -158,6 +158,19 @@ function CodeBasePage(prop) {
...
+      const gitlabRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'gitlab')
+      if (githubRepo !== undefined) {
+        getCommitFromGitHub()
+      }
+      if (gitlabRepo !== undefined) {
+        getCommitFromGitLab()
+      }
+    }
+  }, [isLoading]);
+
```
Comment (VincentETN) [inline src/App/component/CodeBasePage.jsx]: duplicate

**Review by VincentETN — APPROVED** [review]

**Review by vitachu0528 — APPROVED** [review]

**Pull request merged**